File: April 5, 2026.js
function getRotation(n) {

  let rotation = 0;
  const digitCount = String(n).length;
  let newN = n;

  while(true) {

    if (newN % digitCount == 0) {
      return rotation
    }

    let newNStr = String(newN);
    newNStr = newNStr.slice(1) + newNStr[0];
    newN = Number(newNStr);
    rotation += 1;
    if (newN == n) {
      break
    }

  }

  return "none";
}

File: April 5, 2026.py
def get_rotation(n):

    rotation = 0
    new_n = n
    digit_count = len(str(n))

    while True:

        if new_n % digit_count == 0:
            return rotation

        new_n_s = str(new_n)
        new_n_s = new_n_s[1:] + new_n_s[0]
        new_n = int(new_n_s)
        rotation += 1
        if new_n == n:
            break

    return "none"